### PR TITLE
season stats methods

### PR DESCRIPTION
**Pull request by ossar3**
some variables need to be changed but the framework is here for all methods

### CODE REVIEW

File: lib/season_statistics.rb [hunk 1]
```diff
@@ -69,9 +103,23 @@ def fewest_tackles
...
+        tackles_teams = Hash.new(0)
+        @game_data.each do |game|
+            team_id = game[:team_id]
+
+            tackles_teams[team_id] += game[:tackles].to_i
+        end
+           
+            team_most_tackles = tackles_teams.max_by { |team, tackles| tackles }[0]
+    
+            find_team_name(team_most_tackles)
```
Comment (sethverrill) [inline lib/season_statistics.rb:122]: We have a method for team name in the updated stat_tracker you can pull from

File: lib/season_statistics.rb [hunk 2]
```diff
@@ -69,9 +103,23 @@ def fewest_tackles
...
-
+    
+            find_team_name(team_fewest_tackles)
+        end
+        def find_team_name(team_id)
+            #get the corresponding team name to the team id number in teams...
+        end
     def most_tackles
+        tackles_teams = Hash.new(0)
+        @game_data.each do |game|
```
Comment (sethverrill) [inline lib/season_statistics.rb:114]: your game_data will probably have to be changed to @game_teams or @games with the new code

File: lib/season_statistics.rb [hunk 3]
```diff
@@ -8,9 +8,43 @@ def initialize(game_data, team_data)
...
+    end
+
+
+
     def least_accurate_team
+        goals = Hash.new(0)
+        shots = Hash.new(0)
+ 
+        game_teams .each do |game|#gets goals and shots for each team
+         team_id = game[:team_id]
```
Comment (sethverrill) [inline lib/season_statistics.rb:37]: small typo here

File: lib/season_statistics.rb [hunk 4]
```diff
@@ -8,9 +8,43 @@ def initialize(game_data, team_data)
...
+         goals[team_id] += game[:goals].to_i
+         shots[team_id] += game[:shots].to_i
+        end
+        least_accurate_team_id = goals.keys.min_by do |team_id|
+         accuracy(goals[team_id], shots[team_id])#calculates accuracy for each team and then finds most accurate
+        end
+ 
+        find_team_name(least_accurate_team_id_accurate_team_id)
+     end
```
Comment (sethverrill) [inline lib/season_statistics.rb:46]: small typo here

File: lib/season_statistics.rb (hunk 1 above)
Comment (MiTOBrien) [reply lib/season_statistics.rb:122]: This is why it is important to make small updates (aka 1 method) and make sure you do frequent pull requests to your local machine from main.  It keeps you up do date with the code everyone else on the time is writing.

File: lib/season_statistics.rb (hunk 3 above)
Comment (MiTOBrien) [reply lib/season_statistics.rb:37]: a space between game_teams .each

**Comment by MiTOBrien** [comment]
Approving with the understanding that code cleanup/modification is needed.

**Pull request merged**